### PR TITLE
[skia] Add skdescriptor

### PR DESCRIPTION
**Pull request by kjlubick**
Won't work until https://skia-review.googlesource.com/c/skia/+/249776 lands

### CODE REVIEW

**Comment by inferno-chromium** [comment]
ping us here when it lands.

**Comment by kjlubick** [comment]
PTAL: Other cl is in the submit queue

**Comment by Dor1s** [comment]
Re-launched Travis to verify this

File: projects/skia/skdescriptor_deserialize.options [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[libfuzzer]
+max_len = 1024
```
Comment (Dor1s) [inline projects/skia/skdescriptor_deserialize.options]: This is discouraged, please prefer using an explicit size check in the fuzz target source  code (i.e. add `if (size > 1024) return 0;` in the beginning of `LLVMFuzzerTestOneInput`). Not a blocking issue, can be addressed later

Comment (kjlubick) [reply projects/skia/skdescriptor_deserialize.options]: Sounds good - I was unaware of that policy change.

**Comment by Dor1s** [comment]
Travis doesn't seem to be happy with this change.

**Comment by Dor1s** [comment]
Let me land and see if Jenkins survives this

**Comment by Dor1s** [comment]
Because clang errors look like OOMs or other weird issues that are not obviously related to the project code

**Comment by Dor1s** [comment]
The build failed with the following error:

```
2019-10-22T19:25:03.179564207Z Step #4: [839/1819] compile ../../fuzz/oss_fuzz/FuzzSkDescriptorDeserialize.cpp I 
2019-10-22T19:25:03.180019719Z Step #4: FAILED: obj/fuzz/oss_fuzz/skdescriptor_deserialize.FuzzSkDescriptorDeserialize.o  I 
2019-10-22T19:25:03.180027407Z Step #4: clang++ -MD -MF obj/fuzz/oss_fuzz/skdescriptor_deserialize.FuzzSkDescriptorDeserialize.o.d -DNDEBUG -DSK_GAMMA_APPLY_TO_A8 -DSK_ALLOW_STATIC_GLOBAL_INITIALIZERS=1 -DGR_TEST_UTILS=1 -DSK_GL -DSK_SUPPORT_PDF -DSK_HAS_JPEG_LIBRARY -DSK_HAS_HEIF_LIBRARY -DSK_HAS_PNG_LIBRARY -DSK_CODEC_DECODES_RAW -DSK_ENABLE_SKSL_INTERPRETER -DSKVM_JIT -DSK_HAS_WEBP_LIBRARY -DSK_HAS_WUFFS_LIBRARY -DSK_XML -DSK_R32_SHIFT=16 -DSK_SUPPORT_ATLAS_TEXT=1 -I../.. -fPIC -fstrict-aliasing -O3 -fdata-sections -ffunction-sections -g -Wall -Wextra -Winit-self -Wpointer-arith -Wsign-compare -Wvla -Wno-deprecated-declarations -Wno-maybe-uninitialized -fcolor-diagnostics -Weverything -Wno-unknown-warning-option -Wno-nonportable-include-path -Wno-nonportable-system-include-path -Wno-cast-align -Wno-cast-qual -Wno-conversion -Wno-disabled-macro-expansion -Wno-documentation -Wno-documentation-unknown-command -Wno-double-promotion -Wno-exit-time-destructors -Wno-float-equal -Wno-format-nonliteral -Wno-global-constructors -Wno-missing-prototypes -Wno-missing-variable-declarations -Wno-pedantic -Wno-reserved-id-macro -Wno-shadow -Wno-shift-sign-overflow -Wno-signed-enum-bitfield -Wno-switch-enum -Wno-undef -Wno-unreachable-code -Wno-unreachable-code-break -Wno-unreachable-code-return -Wno-unused-macros -Wno-unused-member-function -Wno-unused-template -Wno-zero-as-null-pointer-constant -Wno-thread-safety-negative -Wno-bad-function-cast -Wno-covered-switch-default -Wno-deprecated -Wno-missing-noreturn -Wno-old-style-cast -Wno-padded -Wno-newline-eof -Wno-implicit-fallthrough -Wno-unused-parameter -std=c++14 -fno-exceptions -fno-rtti -Wnon-virtual-dtor -Wno-noexcept-type -Wno-abstract-vbase-init -Wno-weak-vtables -Wno-c++98-compat -Wno-c++98-compat-pedantic -Wno-undefined-func-template -Wno-return-std-move-in-c++11 -O1 -fno-omit-frame-pointer -gline-tables-only -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fsanitize=address -fsanitize-address-use-after-scope -fsanitize=fuzzer-no-link -stdlib=libc++ -Wno-zero-as-null-pointer-constant -Wno-unused-template -Wno-cast-qual -I/src/skia/third_party/externals/swiftshader/include -DGR_EGL_TRY_GLES3_THEN_GLES2 -fno-sanitize=vptr -DIS_FUZZING_WITH_LIBFUZZER -c ../../fuzz/oss_fuzz/FuzzSkDescriptorDeserialize.cpp -o obj/fuzz/oss_fuzz/skdescriptor_deserialize.FuzzSkDescriptorDeserialize.o I 
2019-10-22T19:25:03.180029829Z Step #4: ../../fuzz/oss_fuzz/FuzzSkDescriptorDeserialize.cpp:33:34: error: use of undeclared identifier 'fuzz' I 
2019-10-22T19:25:03.180031412Z Step #4:     FuzzSkDescriptorDeserialize(&fuzz); I 
2019-10-22T19:25:03.180032885Z Step #4:                                  ^ I 
2019-10-22T19:25:03.180034370Z Step #4: 1 error generated. I 
```

**Comment by kjlubick** [comment]
Will land a fix shortly

**Comment by kjlubick** [comment]
Fix landed in Skia repo: https://skia-review.googlesource.com/c/skia/+/250177 Next time the fuzzer rolls forward, it should pick that up.

**Comment by Dor1s** [comment]
Thanks, Kevin! I've scheduled a build manually, should happen in a few hours.

**Comment by Dor1s** [comment]
The build succeeded!

**Comment by kjlubick** [comment]
Thank you for manually pushing and following up.

**Pull request merged**